### PR TITLE
add else case messages

### PR DESCRIPTION
**Pull request by BrainyGuyOrg**

### CODE REVIEW

File: src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala [hunk 1]
```diff
@@ -164,7 +169,9 @@ object SparkExpressionToCExpression {
         Right(StringProducer.copyString(name))
       case Alias(AttributeReference(name, StringType, _, _), name2) =>
         Right(StringProducer.copyString(name))
-
+      case _ => sys.error(s"SparkExpressionToCExpression.evalString: " +
+        s"Could not match ${Try(expression.toString)}, " +
```
Comment (q10) [inline src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala:173]: In general, `toString` should not be `throw`ing, so `Try` may not be needed

Comment (BrainyGuyOrg) [reply src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala:173]: Right. In general. There are some expressions that throw an exception when you try to convert them into a string. I found out about them the hard way (LOL). Not sure what causes it. Recursive expression definitions? Not sure. But the "Try" is definitely needed in some situations.

Comment (wgip) [reply src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala:173]: The `case _ => ... ` should actually be `case other => Left(other)`, I had missed this when writing the code.
The parent already handles the `Left`- side.

**Review by wgip — CHANGES_REQUESTED** [review]

File: src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala [hunk 2]
```diff
@@ -95,6 +97,9 @@ object SparkExpressionToCExpression {
         ar.withName(s"input_${idx}->data[left_out[i]]")
       case idx if (rightIds.contains(ar.exprId)) =>
         ar.withName(s"input_${idx}->data[right_out[i]]")
+      case _ => sys.error(s"SparkExpressionToCExpression.referenceReplacer: " +
```
Comment (wgip) [inline src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala:100]: has this happened? Just wondering.

**Review by wgip — APPROVED** [review]

File: src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala (hunk 1 above)
Comment (BrainyGuyOrg) [reply src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala]: Change made.

File: src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala (hunk 2 above)
Comment (BrainyGuyOrg) [reply src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala:100]: Not sure. But seems like a good check to add.

**Pull request merged**